### PR TITLE
new: working on conference or booth object and event type #FoundationFunding 

### PR DESCRIPTION
**Pull request by DorianVabre**
Specification :  https://wiki.dolibarr.org/index.php/Draft:Module_Event_Organization

This PR :
- [X] Public pages to register (as attendee of a conference)

Next PR for these feature
- [ ] Public pages to register (as speaker or booth responsible) + Add also a link on the ConferenceorBooth object to the payment page /public/payment/newpayment.php to pay a booth (with typeobject and Id of ConferenceorBooth) 
- [ ] Add also a link on the ConferenceOrBoothAttendee object to the payment page /public/payment/newpayment.php to pay a booth (with typeobject and Id of ConferenceOrBoothAttendee) . 
- [ ] New tab on project : Add also button "Send emailing to contact ConferenceOrBooth" => On mass action
- [ ] New tab on project :  Add also button "Send emailing to attendees" => On mass action
- [ ] Add substitution variable for link to ICS file of event + Link to ICS
- [ ] Add a public url to ICS file
- [ ] Enhance the public payment page

Already Done:
- [X] **Module Event Organization**
- [X] Trigger on validation of project
- [X] **New object ConferenceOrBooth**
- [X] **New tab on project**
- [X] **Menus**
- [X] **Permissions**
- [X] New object ConferenceOrBooth : (An attendee will be a prospect third party with a given Tag->cancelled) + Add Contact/Adress Tab with role "Conference Speaker" or "Booth responsible" (Need to create role at module activation)
- [X] ConferenceOrBooth card must be displayed bellow the Event/project Card (like in Project and task tab) 
- [X] Remove fk_project from llx_eventorganization_conferenceorboothattendee table
- [X] **Module Event Organization**
- [X] Add a field "Allow unknown people to suggest conferences" (integer 0 or 1)
- [X] Add a field "Allow unknown people to suggest booth" (integer 0 or 1)
- [X] Add a field "Price of registration" in project
- [X] Add a field "Price of booth" in project
- [X] **New object ConferenceOrBoothAttendee**
- [X] On contact tab on Conference Or Booth. Add Contact/Adress Tab with role "Conference Speaker" or "Booth responsible". Need to create role at module activation

### CODE REVIEW

**Comment by eldy** [comment]
@FHenry is also working on same feature. Florian, is this PR a collaborative work with you or not ?
if not, can i validate without being in conflict with your coming work ?

**Comment by atm-maxime** [comment]
I'd say he's working with @FHenry, he is an IUT student doing traineeship at Scopen according to his LinkedIn profile.

**Comment by DorianVabre** [comment]
> @FHenry is also working on same feature. Florian, is this PR a collaborative work with you or not ?
> if not, can i validate without being in conflict with your coming work ?

> I'd say he's working with @FHenry, he is an IUT student doing traineeship at Scopen according to his LinkedIn profile.

Hello, you are right. This is a collaborative work with @FHenry who supervises me in this work.

**Comment by FHenry** [comment]
@eldy I confirm, I supervise the development from @DorianVabre

File: htdocs/eventorganization/conferenceorbooth_card.php [hunk 1]
```diff
@@ -493,8 +493,17 @@
 	//$keyforbreak='fieldkeytoswitchonsecondcolumn';	// We change column just before this field
 	//unset($object->fields['fk_project']);				// Hide field already shown in banner
 	//unset($object->fields['fk_soc']);					// Hide field already shown in banner
-	include DOL_DOCUMENT_ROOT.'/core/tpl/commonfields_view.tpl.php';
+	$link_subscription = 'http://localhost/dolibarr/htdocs/public/eventorganization/attendee_subscription.php?id=';
```
Comment (eldy) [inline htdocs/eventorganization/conferenceorbooth_card.php:496]: localhost is hard coded.
Use here 
```
global $dolibarr_main_url_root;
$link_subscription = $dolibarr_main_url_root.'/public/...

```

File: htdocs/eventorganization/conferenceorbooth_card.php [hunk 2]
```diff
@@ -493,8 +493,17 @@
...
+	//$link_subscription .= base64_encode($id);
+	$key = 'DV3PH';
+	$link_subscription .= dol_encode($id, $key);
+	$link_subscription .= '&securekey='.$conf->global->EVENTORGANIZATION_SECUREKEY;
```
Comment (eldy) [inline htdocs/eventorganization/conferenceorbooth_card.php:500]: Use
$link_subscription .= '&securekey='.urlencode($conf->global->EVENTORGANIZATION_SECUREKEY);
so if param was corrupted it is not possible to disturb the url forging.

**Pull request merged**